nowledge-mem-codex-plugin/hooks/nmem-stop-save.py

```
@@ -0,0 +1,283 @@
+    import_payload = {
+        "title": derive_thread_title(parsed, cwd),
+        "messages": import_messages,
+    }
+
+    with tempfile.NamedTemporaryFile("w", suffix=".json", delete=False, encoding="utf-8") as handle:
+        json.dump(import_payload, handle, ensure_ascii=False)
+        temp_path = Path(handle.name)
+
+    try:
+        command = [*nmem_command, "--json", "t", "import", "-f", str(temp_path), "--id", thread_id, "-s", "codex"]
+        result = subprocess.run(
```

Ruff (0.15.9)

[error] `subprocess` call: check for execution of untrusted input

(S603)